mods/farming/api.lua

```diff
@@ -200,14 +167,16 @@ farming.grow_plant = function(pos, elapsed)
 
 	-- grow seed
 	if minetest.get_item_group(node.name, "seed") and def.fertility then
-		local soil_node = minetest.get_node_or_nil({x = pos.x, y = pos.y - 1, z = pos.z})
+		local soil_node = minetest.get_node_or_nil(
+			{x = pos.x, y = pos.y - 1, z = pos.z})
 		if not soil_node then
 			tick_again(pos)
 			return
 		end
 		-- omitted is a check for light, we assume seeds can germinate in the dark.
-		for _, v in pairs(def.fertility) do
-			if minetest.get_item_group(soil_node.name, v) ~= 0 then
+		for k = 1, #def.fertility do
+			local groupname = def.fertility[k]
+			if minetest.get_item_group(soil_node.name, groupname) ~= 0 then
```

Comment:
Why? The previous code did the same for less code. In terms of speed it is negligible either way.

Reply:
I've changed it to ipairs now.